docs/developer/architecture/control-plane-ingress.rst

```diff
@@ -0,0 +1,164 @@
+Control Plane Ingress, so we expect the user to provide this Control Plane
+Ingress IP in the Bootstrap configuration file.
+To have some backward compatibility this Ingress IP is only mandatory when
+you use MetalLB and will default to Bootstrap Control Plane IP if not
+(so that we have the same behavior as before).
+
+This Control Plane Ingress IP can be changed at any time just by editing
+the Bootstrap configuration file and follow a simple documented procedure
+with some Salt states to reconfigure every component that needs to be
+re-configured.
+
+NOTE: Changing this Control Plane Ingress IP means we need to reconfigure
+all Kubernetes APIServer since we use this Ingress IP as an OIDC provider.
+
+MetalLB Configuration
+~~~~~~~~~~~~~~~~~~~~~
+
+MetalLB is not deployed in every environment so it needs to be configurable
+from the Bootstrap configuration file, that’s why we have a new field about
+MetalLB in the Control Plane network section.
+
+MetalLB can use 2 different protocols either Layer2 either BGP, by default
+we will use a simple Layer2 configuration for MetalLB so that if the user
+environment allows using Layer2 protocol of MetalLB you just have to enable
+MetalLB, provide an Ingress IP for the Control Plane and that it.
+
+Default MetalLB configuration:
+
+.. code-block:: yaml
+
+    address-pools:
+    - name: ingress-ip
+      protocol: layer2
+      addresses:
+      - <control-plane ingress ip>/32
+      auto-assign: false
+
+But, since BGP router configuration can be really different between various
+environments, we need to make sure this
+`MetalLB configuration <https://metallb.universe.tf/configuration/#bgp-configuration>`_
+is exposed, that’s why we also have the ability to override default MetalLB
+configuration from the Bootstrap configuration file in order to provide BGP
+router information.
+
+Same as the Control Plane Ingress IP, we can switch from non-MetalLB to MetalLB
+(and the opposite) at any time just by following the same procedure.
+
+Deployment
+~~~~~~~~~~
+
+As for every other addon in MetalK8s, we will use the MetalLB helm chart and
+render this one using a specific “option” file. But this one will not be always
+deployed as we only want to deploy it when a specific key is set in the
+Bootstrap configuration file, so in the Salt pillar at the end.
+
+When we use MetalLB we do not want to use the same NGINX Ingress Controller
+deployments, since MetalLB will be the entry point in the Kubernetes cluster
+we do not need to use a DaemonSet running on every Control Plane nodes,
+instead, we will use a Deployment with 2 replicas.
+
+We also need to configure the Service for Ingress Controller differently
+depending on if we use MetalLB or not when we use it we want to use a
+LoadBalancer service and set the LoadBalancerIP to IngressIP provided by
+the user. If we do not use MetalLB then we use ClusterIP Service with IngressIP
+provided by the user as External IPs.
```

Comment:
No mention of `externalTrafficPolicy`?

Reply:
Right, I will add it (btw it's done in implementation PR)